R/lint.R

```diff
@@ -0,0 +1,143 @@
+# Retrieve the lifecycle status defined in each Rd file
+db_lifecycle <- function(db) {
+  lifecycle_patterns <- paste0("(?:",
+    paste(collapse = "|",
+      c("lifecycle::badge\\([\\\\]\"",
+        "rlang:::lifecycle\\([\\\\]\"",
+        "list\\(\"lifecycle-",
+        "https://www.tidyverse.org/lifecycle/#"
+      )),
+    ")([\\w-]+)"
+  )
+
+  desc <- lapply(db, asNamespace("tools")$.Rd_get_metadata, "description")
+  lapply(desc, function(x) {
+    res <- regexpr(lifecycle_patterns, x, perl = TRUE)
+    starts <- attr(res, "capture.start")
+    ends <- starts + attr(res, "capture.length") - 1
+    substring(x, starts, ends)
+  })
+}
+
+# Retrieve the functions listed in the usage for each Rd file in the database
+db_function <- function(db) {
+  usage <- lapply(db, asNamespace("tools")$.Rd_get_section, "usage")
+  lapply(usage, get_usage_function_names)
+}
+
+#' @param package The name of installed packages.
+#' @param which The lifecycle statuses to return for installed packages.
+#'   Include `NA` if you want to include functions without a specified lifecycle
+#'   status in the results.
+#' @export
+#' @rdname lint_lifecycle
+pkg_function_lifecycle <- function(package, which = c("superseded", "deprecated", "questioning", "defunct", "experimental", "soft-deprecated", "retired")) {
+  rlang::check_installed("vctrs")
+  which <- match.arg(which, several.ok = TRUE)
+  stopifnot(is.character(package) && length(package) == 1)
```

Comment:
Also checks for `NA`

```suggestion
  stopifnot(is_string(package))
```